Client/Frontend/Settings/SettingsViewController.swift

```diff
@@ -94,13 +98,54 @@ class SettingsViewController: TableViewController {
     }
     
     override func viewDidLoad() {
+        super.viewDidLoad()
+        
         navigationItem.title = Strings.Settings
         tableView.accessibilityIdentifier = "SettingsViewController.tableView"
         dataSource.sections = sections
         
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(triggerQAMenu))
+        tapGesture.numberOfTapsRequired = 7
+        tapGesture.cancelsTouchesInView = false
+        tapGesture.delaysTouchesEnded = false
+        navigationController?.navigationBar.addGestureRecognizer(tapGesture)
+        
         applyTheme(theme)
     }
     
+    private func displayRewardsDebugMenu() {
+        guard let rewards = rewards else { return }
+        let settings = QASettingsViewController(rewards: rewards)
+        settings.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(tappedQADone))
+        let container = UINavigationController(rootViewController: settings)
+        present(container, animated: true)
+    }
+    
+    @objc private func triggerQAMenu() {
+        guard rewards != nil && navigationController?.visibleViewController === self else { return }
+        
+        let alert = UIAlertController(title: "Nothing to see here", message: "What's the secret password", preferredStyle: .alert)
+        alert.addTextField(configurationHandler: {
+            $0.returnKeyType = .go
+            $0.isSecureTextEntry = true
+            $0.autocorrectionType = .no
+            $0.autocapitalizationType = .none
+        })
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+        alert.addAction(UIAlertAction(title: "Submit", style: .default, handler: { [unowned self] action in
+            guard let textField = alert.textFields?.first else { return }
+            let qaPassword = Bundle.main.infoDictionaryString(forKey: "QA_MENU_PW")
```

Comment:
Is this what Android does too?

Reply:
Yeah basically the same, this is only for accessing it while in prod